incubator/virtualcluster/pkg/syncer/cluster/fake_cluster.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cluster
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	clientset "k8s.io/client-go/kubernetes"
+	clientgocache "k8s.io/client-go/tools/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kubernetes-sigs/multi-tenancy/incubator/virtualcluster/pkg/apis/tenancy/v1alpha1"
+	"github.com/kubernetes-sigs/multi-tenancy/incubator/virtualcluster/pkg/syncer/constants"
+	"github.com/kubernetes-sigs/multi-tenancy/incubator/virtualcluster/pkg/syncer/conversion"
+	"github.com/kubernetes-sigs/multi-tenancy/incubator/virtualcluster/pkg/syncer/mccontroller"
+)
+
+type fakeCluster struct {
+	key           string
+	vc            *v1alpha1.Virtualcluster
+	fakeClientset clientset.Interface
+	fakeClient    client.Client
+}
+
+var _ mccontroller.ClusterInterface = &fakeCluster{}
+
+func NewFakeTenantCluster(vc *v1alpha1.Virtualcluster, fakeClientSet clientset.Interface, fakeClient client.Client) (*fakeCluster, error) {
+	cluster := &fakeCluster{
+		key:           conversion.ToClusterKey(vc),
+		vc:            vc,
```

Comment:
static vc object. If there are any change to vc when testing later, please put vc inside a vclister